sale_automatic_workflow/models/automatic_workflow_job.py

```diff
@@ -116,6 +116,58 @@ def _validate_invoices(self, validate_invoice_filter):
                     invoice.with_company(invoice.company_id), validate_invoice_filter
                 )
 
+    def _do_send_invoice(self, invoice, domain_filter):
+        """Validate an invoice, filter ensure no duplication"""
+        if not self.env["account.move"].search_count(
+            [("id", "=", invoice.id)] + domain_filter
+        ):
+            return "{} {} job bypassed".format(invoice.display_name, invoice)
+
+        # take the context from the actual action_invoice_sent method
+        action = invoice.action_invoice_sent()
+        action_context = action["context"]
+
+        if self.env.su:
+            # sending mail in sudo was meant for it being sent from superuser
+            self = self.with_user(SUPERUSER_ID)
+
+        # Create the email using the wizard
+        invoice_send_wizard = (
+            self.env["account.invoice.send"]
+            .with_context(
+                action_context,
+                mark_invoice_as_sent=True,
+                active_ids=[invoice.id],
+                force_email=True,
+            )
+            .create(
+                {
+                    "is_print": False,
+                    "composition_mode": "comment",
+                    "model": "account.move",
+                    "res_id": invoice.id,
+                }
+            )
+        )
+
+        invoice_send_wizard._compute_composition_mode()
+        invoice_send_wizard.onchange_template_id()
```

Comment:
AFAIU, you don't need to call these as long as you call `onchange_is_email` below.

Reply:
I removed it, thanks!